### PR TITLE
May cleanup 1

### PR DESCRIPTION
**Pull request by russgold**
Added DomainPresence tests, improved reliability of tests, improved code readability

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/Watcher.java [hunk 1]
```diff
@@ -87,11 +87,15 @@ void start(ThreadFactory factory) {
...
 
-    while (!isDraining()) {
-      if (isStopping()) setIsDraining(true);
-      else watchForEvents();
+      while (!isDraining()) {
+        if (isStopping()) setIsDraining(true);
+        else watchForEvents();
+      }
+    } catch (Exception e) {
+      LOGGER.warning("Exception caught in thread " + Thread.currentThread().getName(), e);
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/Watcher.java]: INFO and above need to be internationalized.

File: operator/src/main/java/oracle/kubernetes/operator/helpers/DomainPresenceInfoManager.java [hunk 2]
```diff
@@ -24,9 +22,12 @@ public static DomainPresenceInfo getOrCreate(String ns, String domainUID) {
   }
 
   public static DomainPresenceInfo getOrCreate(Domain domain) {
+    DomainPresenceInfo oldInfo = lookup(domain.getSpec().getDomainUID());
+    DomainPresenceInfo oldInfo2 = domains.get(domain.getSpec().getDomainUID());
     DomainPresenceInfo createdInfo = new DomainPresenceInfo(domain);
     DomainPresenceInfo existingInfo =
         domains.putIfAbsent(domain.getSpec().getDomainUID(), createdInfo);
+    if (existingInfo == null && oldInfo != null) System.out.println("This can't be");
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/DomainPresenceInfoManager.java]: Odd...  can we use a different pattern than System.out?  throw new IllegalStateException()?

Comment (russgold) [reply operator/src/main/java/oracle/kubernetes/operator/helpers/DomainPresenceInfoManager.java]: Um... that was a debug statement. It doesn't belong there at all. Oops.

File: operator/src/main/java/oracle/kubernetes/operator/Watcher.java (hunk 1 above)
Comment (russgold) [reply operator/src/main/java/oracle/kubernetes/operator/Watcher.java]: Debug code I forgot to remove

**Pull request merged**